### PR TITLE
System default theme added Fixes #1397

### LINKED ISSUE #1397: Option to use system default theme and respect Android values
### Describe the bug

Android 10 allows you to toggle system theme between light and dark and Android 11 added option to schedule theme change eg. you can set light theme during the day and dark theme during the evening.

One would expect DuckDuckGo browser to toggle and change themes after you change something in Android 10+ settings however that is not possible because there is only manual toggle in the app and not the option to honor and follow system defaults.

### How to Reproduce

On android 10:

- go to "Settings"
- go to "Screen"
- Toggle "Dark theme"

### Expected behavior

One would expect DuckDuckGo browser to toggle and change themes however that is not possible because there is only manual toggle in the app and not the option to honor system defaults.

### Environment

```markdown
- DDG App Version: 5.94.3 (59403)
- Device: Motorola One Fusion+
- OS: Android 10
```

### PR DESCRIPTION
**Pull request by 4shutosh**
Task/Issue URL: #1397

**Description**:
Added system default theme option for supporting devices Android 10 onwards

**Steps to test this PR**:
1. Open Settings
2. Open theme group
3. Devices above 10 and above will have an option to set the System Default theme option.

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

**Comment by 4shutosh** [comment]
Doubts: 
- How do I add all the string values of other language translations, is there any efficient option other than adding them manually?
- Will need the Pixel param for the new SYSTEM_DEFAULT click statistics `AppPixelName:SETTINGS_THEME_TOGGLED_SYSTEM_DEFAULT`

**Comment by 4shutosh** [comment]
I've added translations as well.

Here is a screenshot of the entire implementation:
sampleSS

Let me know with any changes, Please Review :)

**Comment by 4shutosh** [comment]
Video Sample for the same:
videoSample

**Comment by malmstein** [comment]
@4shutosh thanks for the contribution! We are discussing internally how we want to present the Light / Dark / System options, but it's looking good so far. 

I'll get back to you with the next steps regarding translations, etc...

Good work!

**Comment by 4shutosh** [comment]
Sure @malmstein, please let me know of any changes required.

File: app/src/main/java/com/duckduckgo/app/pixels/AppPixelName.kt [hunk 1]
```diff
@@ -111,6 +111,7 @@ enum class AppPixelName(override val pixelName: String) : Pixel.PixelName {
     SETTINGS_OPENED("ms"),
     SETTINGS_THEME_TOGGLED_LIGHT("ms_tl"),
     SETTINGS_THEME_TOGGLED_DARK("ms_td"),
+    SETTINGS_THEME_TOGGLED_SYSTEM_DEFAULT("ms_tsd"), // confirm? the code here for statistics
```
Comment (malmstein) [inline app/src/main/java/com/duckduckgo/app/pixels/AppPixelName.kt:114]: this is good, just remove the comment

File: app/src/main/res/layout/content_settings_general.xml [hunk 2]
```diff
@@ -29,14 +29,15 @@
         app:layout_constraintStart_toStartOf="parent"
         app:layout_constraintTop_toTopOf="parent" />
 
-    <androidx.appcompat.widget.SwitchCompat
-        android:id="@+id/lightThemeToggle"
-        style="@style/SettingsSwitch"
-        android:text="@string/settingsLightTheme"
-        android:theme="@style/SettingsSwitchTheme"
+    <com.duckduckgo.app.settings.SettingsOptionWithSubtitle
+        android:id="@+id/themeToggle"
```
Comment (malmstein) [inline app/src/main/res/layout/content_settings_general.xml:33]: can you rename this to `selectedThemeSetting` to follow convention? It's not really a toggle anymore.

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -151,6 +151,9 @@
     <string name="settingsHeadingOther">Other</string>
     <string name="settingsHeadingPrivacy">Privacy</string>
     <string name="settingsLightTheme">Light Theme</string>
+    <string name="settingsTheme">Theme</string>
```
Comment (malmstein) [inline app/src/main/res/values/strings.xml:154]: new translations go into `settings-untranslated.xml` and only in English

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -151,6 +151,9 @@
...
+    <string name="settingsDarkTheme">Dark Theme</string>
```
Comment (malmstein) [inline app/src/main/res/values/strings.xml:155]: ditto, also let's just have `Dark` and remove `Theme`

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -151,6 +151,9 @@
...
+    <string name="settingsSystemTheme">System Default</string>
```
Comment (malmstein) [inline app/src/main/res/values/strings.xml:156]: ditto

File: app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -493,6 +496,9 @@
     <string name="disableLoginDetectionDialogNegative">CANCEL</string>
     <string name="fireproofWebsiteRemovalConfirmation">Fireproofing removed for &lt;b>%s&lt;/b></string>
 
+    <!--    Theme Settings-->
```
Comment (malmstein) [inline app/src/main/res/values/strings.xml:499]: ditto

File: app/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -493,6 +496,9 @@
...
+    <string name="settingsThemeDialogSave">Set Theme</string>
```
Comment (malmstein) [inline app/src/main/res/values/strings.xml:500]: ditto

File: app/src/main/res/layout/settings_theme_selector_fragment.xml [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="8dp"
+            android:padding="10dp"
+            android:paddingStart="10dp"
+            android:paddingEnd="10dp"
+            android:text="@string/settingsDarkTheme"/>
+
+        <RadioButton
+            android:id="@+id/themeSelectorSystemDefault"
```
Comment (malmstein) [inline app/src/main/res/layout/settings_theme_selector_fragment.xml:56]: can you move System Default to be the first item in the dialog?

File: common-ui/src/main/java/com/duckduckgo/mobile/android/ui/Theming.kt [hunk 9]
```diff
@@ -30,7 +33,8 @@ import com.duckduckgo.mobile.android.ui.Theming.Constants.THEME_MAP
 
 enum class DuckDuckGoTheme {
     DARK,
-    LIGHT;
+    LIGHT,
+    SYSTEM_DEFAULT;
```
Comment (malmstein) [inline common-ui/src/main/java/com/duckduckgo/mobile/android/ui/Theming.kt:37]: `SYSTEM_DEFAULT` should be the first in the list

File: app/src/main/res/values/strings.xml [hunk 10]
```diff
@@ -151,6 +151,9 @@
     <string name="settingsHeadingOther">Other</string>
     <string name="settingsHeadingPrivacy">Privacy</string>
     <string name="settingsLightTheme">Light Theme</string>
```
Comment (malmstein) [inline app/src/main/res/values/strings.xml:153]: Can you rename this to only `Light`?

File: app/src/main/java/com/duckduckgo/app/settings/SettingsThemeSelectorFragment.kt [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+class SettingsThemeSelectorFragment : DialogFragment() {
+
+    interface Listener {
+        fun onThemeSelected(selectedTheme: DuckDuckGoTheme)
+    }
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+
+        val currentOption: DuckDuckGoTheme =
+            arguments?.getSerializable(DEFAULT_OPTION_EXTRA) as DuckDuckGoTheme? ?: SYSTEM_DEFAULT
```
Comment (malmstein) [inline app/src/main/java/com/duckduckgo/app/settings/SettingsThemeSelectorFragment.kt:40]: the default here should be `LIGHT`

**Review by malmstein — CHANGES_REQUESTED** [review]
Thanks for this @4shutosh!

I've left a few comments, but mostly:

- Delete all the new added translations to different languages. We have an internal process to deal with them
- Move the new added English strings to `strings-untranslated.xml`
- Change order of dialog so System Default is the first item on the list.

**Comment by 4shutosh** [comment]
On it

**Comment by 4shutosh** [comment]
- I've removed translations, also moved `settingsLightTheme` string into the untranslated as the string value was changed to "Theme"
- Made System Default on the top of the list

File: app/src/main/res/values/string-untranslated.xml [hunk 12]
```diff
@@ -40,4 +40,13 @@
     <!-- Email -->
     <string name="emailNotSupported">Device not supported</string>
     <string name="emailNotSupportedExplanation">Email Protection allows you to create private email addresses that remove email trackers. We need to encrypt and store these addresses you create locally on your device. Because your device does not support encrypted storage, Email Protection is unavailable.</string>
+
+    <!-- Settings Activity -->
+    <string name="settingsLightTheme">Light</string>
+    <string name="settingsTheme">Theme</string>
+    <string name="settingsDarkTheme">Dark </string>
```
Comment (malmstein) [inline app/src/main/res/values/string-untranslated.xml:47]: trailing spance

**Comment by malmstein** [comment]
@4shutosh I left a comment about a space added that's not needed. Also, seems like there are some compilation errors:

```
> Task :app:compilePlayDebugAndroidTestKotlin FAILED
e: /Users/runner/work/Android/Android/app/src/androidTest/java/com/duckduckgo/app/settings/SettingsViewModelTest.kt: (148, 20): Unresolved reference: onLightThemeToggled
e: /Users/runner/work/Android/Android/app/src/androidTest/java/com/duckduckgo/app/settings/SettingsViewModelTest.kt: (159, 16): Unresolved reference: onLightThemeToggled
e: /Users/runner/work/Android/Android/app/src/androidTest/java/com/duckduckgo/app/settings/SettingsViewModelTest.kt: (167, 20): Unresolved reference: onLightThemeToggled
e: /Users/runner/work/Android/Android/app/src/androidTest/java/com/duckduckgo/app/settings/SettingsViewModelTest.kt: (179, 16): Unresolved reference: onLightThemeToggled
e: /Users/runner/work/Android/Android/app/src/androidTest/java/com/duckduckgo/app/settings/SettingsViewModelTest.kt: (187, 20): Unresolved reference: onLightThemeToggled
```

Once those are fixed we are ready to approve. Let me know if you need any help!

**Comment by 4shutosh** [comment]
@malmstein I've modified the tests. Please check

I've noticed one thing while implementing this that the `DialogFragments` do not follow the dark theme for some reason. I would like to open an issue for this and work on the same.

**Comment by 4shutosh** [comment]
Another doubt: There is a `darkThemeSwitch` inside AppComponentsActivity inside the Internal Features, does that require any modification as well?

**Comment by 4shutosh** [comment]
Is there anything else required to be done from my side @malmstein? Please let me know

**Review by malmstein — APPROVED** [review]

**Comment by malmstein** [comment]
It's all good now @4shutosh. Thanks for putting in the work!

**Comment by 4shutosh** [comment]
It's my pleasure. Thank you for you help 🙌🏻

**Pull request merged**